build/i2c/src/lib.rs

```diff
@@ -200,9 +200,132 @@ struct I2cSensors {
     names: Option<Vec<String>>,
 }
 
-impl I2cSensors {
-    fn num_sensors(&self) -> usize {
-        self.temperature + self.power + self.current + self.voltage + self.speed
+#[derive(Debug)]
+struct I2cSensorsDescription {
+    // In all multimaps below, the value is the sensor ID. The same sensor ID
+    // can show up in multiple (including all!) of these maps.
+    //
+    // All sensors are guaranteed to be present in `bydevice` and `bykind`, but
+    // may not be present in the other maps (devices may or may not have a
+    // name/bus in app.toml).
+
+    // key: (device, kind)
+    bydevice: MultiMap<(String, Sensor), usize>,
```

Comment:
Two comments on this:
1) It would be easier to read as `by_device` instead of `bydevice`, IMO
2) I think the tuples have outlived their welcome, and we should instead make key structs, e.g.
```rust
#[derive(Eq, PartialEq, Hash, Ord, PartialOrd)] // not sure what MultiMap needs
struct DeviceKey {
    device: String,
    kind: String,
}
```

Reply:
All fair comments - I was trying to change as little as I could here to make sure I wasn't breaking anything, but I'll make a cleanup pass with your suggestions.

Reply:
1 changed by 4e2716cd
2 implemented in 5bf5ae8b